api/odl.py

```diff
@@ -268,8 +300,7 @@ def checkin(self, patron, pin, licensepool):
             raise NotCheckedOut()
         elif status == self.ACTIVE_STATUS:
             # This loan has already been fulfilled, so it needs to be returned through the DRM system.
-            # Do nothing.
-            return
+            raise CannotReturn(_("This loan has already been fulfilled, so it must be returned through the DRM system."))
```

Comment:
Is this always the case or does it depend on the DeliveryMechanism of the LicensePoolDeliveryMechanism? That is, if the book was delivered without DRM it should be returnable through the return link.

Reply:
I don't think a book would ever have "active" status if it didn't have DRM - the DRM system notifies the distributor when the book is fulfilled and that makes it active. Without DRM the distributor wouldn't know and it would probably stay "ready".

Reply:
I could be wrong though - I'll have to check on this.

Reply:
I think maybe the best thing to do here would be to always try the return link for "active" or "ready" license if there is one, but raise an exception if the status doesn't change.